src/Agent.Worker/Build/BuildJobExtension.cs

```diff
@@ -188,6 +188,7 @@ public void ConvertLocalPath(IExecutionContext context, string localPath, out st
             }
 
             executionContext.Debug($"Sync source for endpoint: {SourceEndpoint.Name}");
+            SourceProvider.SetVariablesInEndpoint(executionContext, SourceEndpoint);
```

Comment:
Agree. We should clone the endpoints as soon as we enter the extensions and avoid making any changes to the endpoint in context. WIll make this change